src/rasterstats/main.py

```diff
@@ -146,6 +150,8 @@ def gen_zonal_stats(
         features_iter = read_features(vectors, layer)
         for _, feat in enumerate(features_iter):
             geom = shape(feat['geometry'])
+            if properties:
+                prop_dic = {k: feat['properties'][k] for k in properties}
```

Comment:
What about an alternative design here:

Instead of specifying a list of properties, we could pass the entire properties dict by default. This would break existing user-defined functions so we'd need to catch that below (as you've already done)